### PR TITLE
Make pieces move

### PR DESCRIPTION
**Pull request by chaudhary27**
*pair programmed with @NehemiahAndrews to get this up*

There is one issue that needs to be fixed. Pieces are not tied to a game in context of move piece logic. Our hunch is that in **pieces_controller** `move` method line `selected_piece = Piece.where(is_selected: true, user_id: current_user.id).first` is selecting all the pieces for a given user across games. During the actual move process it messes up a lot of things. We are investigating how to default that line so it knows to only deal with the pieces in context of a game that is in process.

if you have any comments/ideas how to achieve, please share :small_red_triangle:

### CODE REVIEW

**Comment by vincentchin** [comment]
After adding a way to check if moving to a specific destination is a valid move, the original problem where it messes up the board is still persisting. This is caused by the fact that moving the selected piece to a new destination alters the **is_selected* value to false. The redirection is then wrong because selected_piece is looking for the next piece where is_selected is true and has the same user_id (this is where it redirects to another game).

TL:DR We need to make selected_piece look something like this but I haven't figured out how to get the current game.

```ruby
  selected_piece = Piece.where(is_selected: true, user_id: current_user.id, 
game_id: current_game.id).first
```

After that we can add capture logic.

**Comment by NehemiahAndrews** [comment]
Pulled into local environment, verified that pieces move where they are told. Merging.
